src/master/filesystem_node_types.h

```diff
@@ -203,17 +206,19 @@ struct FSNodeDevice : public FSNode {
  * Avg size (10 files) ~ 280B (28B per file)
  */
 struct FSNodeDirectory : public FSNode {
-#if defined(SAUNAFS_HAVE_64BIT_JUDY) && !defined(DISABLE_JUDY_FOR_ENTRIESCONTAINER)
-	typedef judy_map<hstorage::Handle, FSNode *> EntriesContainer;
-#else
 	struct HandleCompare {
-		bool operator()(const hstorage::Handle &a, const hstorage::Handle &b) const {
-			return a.data() < b.data();
+		bool operator()(const std::pair<hstorage::Handle, FSNode *> &a,
+		                const std::pair<hstorage::Handle, FSNode *> &b) const {
+			return std::make_pair(a.first.data(), a.second) <
+			       std::make_pair(b.first.data(), b.second);
 		}
 	};
-	typedef flat_map<hstorage::Handle, FSNode *, std::vector<std::pair<hstorage::Handle, FSNode *>>,
-	HandleCompare> EntriesContainer;
-#endif
+
+	typedef __gnu_pbds::tree<std::pair<hstorage::Handle, FSNode *>,
+	                         __gnu_pbds::null_type, HandleCompare,
+	                         __gnu_pbds::rb_tree_tag,
+	                         __gnu_pbds::tree_order_statistics_node_update>
+	    EntriesContainer;
```

Comment:
How is the portability of this extensions? Are they available only for GCC or also in Clang (for instance)?